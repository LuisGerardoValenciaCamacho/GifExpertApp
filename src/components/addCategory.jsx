import { useState } from "react";
import PropTypes from "prop-types";

export const Addcategory = ({onNewCategory}) => {
    const [inputValue, setInputValue] = useState("");

    const onInputChange = (event) => {
        setInputValue(event.target.value);
    }

    const onSubmit = (event) => {
        event.preventDefault();
        const newInputValue = inputValue.trim()
        if(newInputValue.length <= 1) return;
        onNewCategory(newInputValue)
        setInputValue("");
    }
    
    return (
        <form onSubmit={onSubmit} aria-label="form">
            <input 
                type="text"
                placeholder="Buscar en Gifs" 
                value={inputValue} 
                onChange={onInputChange}
            />
        </form>
    )
}

Addcategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired
}