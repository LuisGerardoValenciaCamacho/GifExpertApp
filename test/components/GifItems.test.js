import React from "react";
import "@testing-library/react";
import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/gifItem";

describe("Pruebas hacia <GifItems/>", () => {

    const title = "Kimetsu";
    const url = "https://kimetsu.com/tanjiro.jpg"

    test("Debe hacer match con el snapshot", () => {
        const {container} = render(<GifItem title={title} url={url}/>)
        expect(container).toMatchSnapshot();
    })
    
    test("Debe mostrar la imagen con el url y el alt indicado", () => {
        render(<GifItem title={title} url={url}/>)
        const {src, alt} = screen.getByRole("img");
        expect(src).toBe(url);
        expect(alt).toBe(title);
    })
    
    test("Debe de mostrar el titulo en el componente", () => {
        render(<GifItem title={title} url={url}/>);
        expect(screen.getByText(title)).toBeTruthy();
    })
})