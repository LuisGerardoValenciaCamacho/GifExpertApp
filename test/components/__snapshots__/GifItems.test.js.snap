// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas hacia <GifItems/> Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Kimetsu"
      src="https://kimetsu.com/tanjiro.jpg"
    />
    <p>
      Kimetsu
    </p>
  </div>
</div>
`;
